onyo/lib/tests/test_commands_mv.py

```diff
@@ -69,12 +70,79 @@ def test_onyo_mv_into_self(inventory: Inventory) -> None:
                   destination=asset_path.parent,
                   message="some subject\n\nAnd a body")
 
+    # source does not exist
+    pytest.raises(ValueError,
+                  onyo_mv,
+                  inventory,
+                  source=inventory.root / "not-existent",
+                  destination=dir_path,
+                  message="some subject\n\nAnd a body")
+
+
+@pytest.mark.ui({'yes': True})
+def test_onyo_mv_errors_before_mv(inventory: Inventory) -> None:
+    """`onyo_mv` must raise the correct error and is not allowed to move/commit anything, if one of
+    the sources does not exist.
+    """
+    asset_path = inventory.root / "somewhere" / "nested" / "TYPE_MAKER_MODEL.SERIAL"
+    destination_path = inventory.root / 'empty'
+    old_hexsha = inventory.repo.git.get_hexsha()
+
+    # one of multiple sources does not exist
+    pytest.raises(ValueError,
+                  onyo_mv,
+                  inventory,
+                  source=[asset_path, inventory.root / "not-existent"],
+                  destination=destination_path,
+                  message="some subject\n\nAnd a body")
+
+    # nothing was moved and no new commit was created
+    assert asset_path.is_file()
+    assert not (destination_path / asset_path.name).is_file()
+    # git is correct
+    assert inventory.repo.git.get_hexsha() == old_hexsha
+    # TODO: verifying cleanness of worktree does not work,
+    #       because fixture returns inventory with untracked stuff
+    # assert inventory.repo.git.is_clean_worktree()
+
+
+@pytest.mark.skip(reason="still a BUG: #516")
+@pytest.mark.ui({'yes': True})
+@pytest.mark.repo_dirs("a/b/c", "a/d/c")
+def test_onyo_mv_possible_error(inventory: Inventory) -> None:
+    """Allow to move a directory into another one with the same name."""
+    source_path = inventory.root / "a" / "b" / "c"
+    destination_path = inventory.root / "a" / "d" / "c"
+    old_hexsha = inventory.repo.git.get_hexsha()
+
+    # move an asset and a dir to the same destination
+    onyo_mv(inventory,
+            source=source_path,
+            destination=destination_path,
+            message="some subject\n\nAnd a body")
+
+    # source
+    assert not source_path.exists()
+    assert (source_path / OnyoRepo.ANCHOR_FILE_NAME) not in inventory.repo.git.files
+    # directory was moved
+    assert (destination_path / source_path.name).is_dir()
+    assert (destination_path / source_path.name / OnyoRepo.ANCHOR_FILE_NAME).is_file()
+    assert inventory.repo.is_inventory_dir(destination_path / source_path.name)
+    assert (destination_path / source_path.name / OnyoRepo.ANCHOR_FILE_NAME) in inventory.repo.git.files
+    # git is correct
```

Comment:
Hm. This comment says nothing really. What's a correct git? ;-)
As far as I'm concerned, no comment needed, but if you want one here, then it should be more clear.
"exactly one commit added" or something like this.